src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -138,43 +127,21 @@ protected override void Dispose(bool disposing)
             }
         }
 
-        private async Task CollectorTask(CancellationToken token, int collectionPeriodMilliseconds, MetricProcessor[] processors)
+        private MetricItem Collect()
```

Comment:
MeterProvider has this method, but it never calls this. Its called by MetricProcessors, whenever they want. Typically in fixed time intervals (PushExporter) or in response to Scrape requests (PullExporter). Eitherway, MeterProvider doesn't care.